src/scraping/collector.py

```
@@ -0,0 +1,123 @@
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        if self.session:
+            await self.session.close()
+
+    async def collect_leads(self, criteria: Dict[str, Any]) -> List[Dict[str, Any]]:
+        """Collect leads based on search criteria"""
+        leads = []
+        try:
+            raw_data = await self._fetch_listings(criteria)
+            leads = self._process_listings(raw_data)
+        except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)